### PR TITLE
dae: add missing subcommand `trace`

### PR DESCRIPTION
**Pull request by oluceps**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Aleksanaa** [comment]
Could you explain in detail (preferably in the commit message) what is the reason for this change?

**Comment by oluceps** [comment]
> Could you explain in detail (preferably in the commit message) what is the reason for this change?

This will build the `trace` command which was added few weeks ago.
```
> ./dae -h
dae is a high-performance transparent proxy solution.

Usage:
  dae [command]

Available Commands:
  export      To export some information for UI developers.
  help        Help about any command
  honk        Let dae call for you.
  reload      To reload config file without interrupt connections.
  run         To run dae in the foreground.
  suspend     To suspend dae. This command puts dae into no-load state. Recover it by 'dae reload'.
  trace       To trace traffic
  validate    To validate dae config.

Flags:
  -h, --help      help for dae
  -v, --version   version for dae

Use "dae [command] --help" for more information about a command.
```

`-tag=trace` will be added when `make dae` and release, but lacks in nixpkgs' packaging before.

**Comment by zzzsyyy** [comment]
This dosen't correctly apply `ldflags` to the `go build`

**Comment by oluceps** [comment]
> This dosen't correctly apply `ldflags` to the `go build`

Yeah, the `ldflags` is now actually redundant. I've removed it.

File: pkgs/tools/networking/dae/default.nix [hunk 1]
```diff
@@ -21,17 +22,11 @@ buildGoModule rec {
...
-  ldflags = [
-    "-s"
-    "-w"
-    "-X github.com/daeuniverse/dae/cmd.Version=${version}"
-    "-X github.com/daeuniverse/dae/common/consts.MaxMatchSetLen_=64"
-  ];
-
-  preBuild = ''
+  buildPhase = ''
     make CFLAGS="-D__REMOVE_BPF_PRINTK -fno-stack-protector -Wno-unused-command-line-argument" \
```
Comment (Aleksanaa) [inline pkgs/tools/networking/dae/default.nix:26]: runHook preBuild
runHook postBuild

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**